[Bug]: Linter execution does not report the issues

### Bug description

Github actions linter detects the issues but failed to put PR comment, when PR is created from forked repository.
There are also silently failed, but job is finished successfully, the linter execution.
We can observe the log in the inter job of PR# https://github.com/miurahr/aqtinstall/actions/runs/12384038278/job/34567887945

### Expected behavior

Successfully put PR comment.

### aqt and python version

N.A.

### Operating System

Linux/Unix

### Relevant log output

```shell
time=2024-12-18T00:40:06.471Z level=INFO msg="reviewdog: [start] runner=pylint"
time=2024-12-18T00:40:06.472Z level=INFO msg="reviewdog: [start] runner=flake8"
time=2024-12-18T00:40:06.472Z level=INFO msg="reviewdog: [start] runner=mypy"
time=2024-12-18T00:40:10.588Z level=INFO msg="reviewdog: [finish] runner=flake8"
time=2024-12-18T00:40:13.991Z level=INFO msg="reviewdog: [finish] runner=mypy\terror=exit status 1"
time=2024-12-18T00:40:16.782Z level=INFO msg="reviewdog: [finish] runner=pylint\terror=exit status 30"
aqt/metadata.py:279:24: [mypy] error: Call to untyped function "Version" in typed
aqt/metadata.py:575:23: [mypy] error: Call to untyped function "Version" in typed
aqt/metadata.py:1012:69: [mypy] error: Call to untyped function "Version" in typed
Error: [mypy] reported by reviewdog 🐶
error: Call to untyped function "Version" in typed

Raw Output:
aqt/metadata.py:279:24: error: Call to untyped function "Version" in typed
Error: [mypy] reported by reviewdog 🐶
error: Call to untyped function "Version" in typed

Raw Output:
aqt/metadata.py:575:23: error: Call to untyped function "Version" in typed
Error: [mypy] reported by reviewdog 🐶
error: Call to untyped function "Version" in typed

Raw Output:
aqt/metadata.py:1012:69: error: Call to untyped function "Version" in typed
aqt/metadata.py:279: [pylint] Line too long (101/100) (line-too-long)
aqt/metadata.py:279:8: [pylint] Unnecessary "else" after "return", remove the "else" and de-indent the code inside it (no-else-return)
aqt/metadata.py:574:4: [pylint] Missing function or method docstring (missing-function-docstring)
aqt/metadata.py:576:12: [pylint] Unnecessary "else" after "return", remove the "else" and de-indent the code inside it (no-else-return)
Error: [pylint] reported by reviewdog 🐶
Line too long (101/100) (line-too-long)

Raw Output:
aqt/metadata.py:279:0: C0301: Line too long (101/100) (line-too-long)
Error: [pylint] reported by reviewdog 🐶
Unnecessary "else" after "return", remove the "else" and de-indent the code inside it (no-else-return)

Raw Output:
aqt/metadata.py:279:8: R1705: Unnecessary "else" after "return", remove the "else" and de-indent the code inside it (no-else-return)
Error: [pylint] reported by reviewdog 🐶
Missing function or method docstring (missing-function-docstring)

Raw Output:
aqt/metadata.py:574:4: C0116: Missing function or method docstring (missing-function-docstring)
Error: [pylint] reported by reviewdog 🐶
Unnecessary "else" after "return", remove the "else" and de-indent the code inside it (no-else-return)

Raw Output:
aqt/metadata.py:576:12: R1705: Unnecessary "else" after "return", remove the "else" and de-indent the code inside it (no-else-return)
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/.github/workflows/check.yml
+++ b/.github/workflows/check.yml
@@ -1,4 +1,4 @@
-name: Check tox tests, lint and types
+name: Check tox tests
 
 on:
   push:

--- a/.github/workflows/run-linter.yml
+++ b/.github/workflows/run-linter.yml
@@ -13,6 +13,8 @@ jobs:
   linter:
     name: Linter
     runs-on: ubuntu-24.04
+    permissions:
+      pull-requests: write
     steps:
       - uses: actions/checkout@v4
         with:
@@ -30,8 +32,6 @@ jobs:
           pip install -U pip tox wheel setuptools coveralls coverage[toml]
           pip install .[check]
       - name: run reviewdog
-        permissions:
-          pull-requests: write
         run: |
           ./bin/reviewdog -runners=flake8,mypy,pylint -reporter=github-pr-review -conf=.reviewdog.yml
         env:

--- a/.github/workflows/run-linter.yml
+++ b/.github/workflows/run-linter.yml
@@ -3,7 +3,11 @@ name: Run linter
 on:
   pull_request:
     branches:
-      - 'master'
+      - master
+    types:
+      - opened
+      - synchronize
+      - reopened
 
 jobs:
   linter:

--- a/.github/workflows/run-linter.yml
+++ b/.github/workflows/run-linter.yml
@@ -2,6 +2,8 @@ name: Run linter
 
 on:
   pull_request:
+    branches:
+      - 'master'
 
 jobs:
   linter:
@@ -24,8 +26,10 @@ jobs:
           pip install -U pip tox wheel setuptools coveralls coverage[toml]
           pip install .[check]
       - name: run reviewdog
+        permissions:
+          pull-requests: write
         run: |
-          ./bin/reviewdog -runners=flake8,mypy,pylint -reporter=github-pr-check -conf=.reviewdog.yml
+          ./bin/reviewdog -runners=flake8,mypy,pylint -reporter=github-pr-review -conf=.reviewdog.yml
         env:
           REVIEWDOG_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 

--- a/.github/workflows/run-linter.yml
+++ b/.github/workflows/run-linter.yml
@@ -33,7 +33,7 @@ jobs:
           pip install .[check]
       - name: run reviewdog
         run: |
-          ./bin/reviewdog -runners=flake8,mypy,pylint -reporter=github-pr-review -conf=.reviewdog.yml
+          ./bin/reviewdog -runners=flake8,mypy -reporter=github-pr-review -conf=.reviewdog.yml
         env:
           REVIEWDOG_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 

